lasagne/layers/conv.py

```diff
@@ -33,29 +33,27 @@ def conv_output_length(input_length, filter_size,
     stride : int
         The stride of the convolution operation.
 
-    border_mode : str, 'valid', 'full', 'same' or 'pad'
-        A string indicating the convolution border mode.
+    pad : int, tuple of int, 'full' or 'same' (default: 0)
+        By default, the convolution is only computed where the input and the
+        filter fully overlap (a valid convolution). When ``stride=1``, this
+        yields an output that is smaller than the input by ``filter_size - 1``.
+        The `pad` argument allows to implicitly pad the input with zeros,
+        extending the output size.
 
-        If 'valid', it is assumed that the convolution is only computed where
-        the input and the filter fully overlap.
+        A single integer results in symmetric zero-padding of the given size on
+        all borders, a tuple of two integers allows different symmetric padding
+        per dimension.
 
-        If 'full', it is assumed that the convolution is computed wherever the
-        input and the filter overlap by at least one position.
-
-        If 'same', it is assumed that the convolution is computed wherever the
-        input and the filter overlap by at least half the filter size, when the
-        filter size is odd. In practice, the input is zero-padded with half the
-        filter size at the beginning and half at the end (or one less than half
-        in the case of an even filter size). This results in an output length
-        that is the same as the input length (for both odd and even filter
-        sizes).
+        ``'full'`` pads with one less than the filter size on both sides. This
+        is equivalent to computing the convolution wherever the input and the
+        filter overlap by at least one position.
 
-        If 'pad', zero padding of `pad` positions is assumed to be applied to
-        the input, and then a valid convolution is applied.
+        ``'same'`` pads with half the filter size on both sides (one less on
+        the second side for an even filter size). When ``stride=1``, this
+        results in an output size equal to the input size.
 
-    pad : int, optional (default 0)
-        If `border_mode` is set to 'pad', this is the size of the padding that
-        is applied on both sides of the input. Otherwise, this is ignored.
+        Note that ``'full'`` and ``'same'`` can be faster than equivalent
+        integer values due to optimizations by Theano.
```

Comment:
This doesn't affect the speed of `conv_output_length` at all, though, so this last paragraph should be removed.